vic/drivers/classic/src/parse_output_info.c

```diff
@@ -31,79 +31,126 @@
  * @brief    Get output info from global parameter file.
  *****************************************************************************/
 void
-parse_output_info(FILE                  *gp,
-                  out_data_file_struct **out_data_files,
-                  out_data_struct       *out_data)
+parse_output_info(FILE                *gp,
+                  stream_struct      **output_streams,
+                  stream_file_struct **out_data_files)
 {
-    extern option_struct options;
-
-    char                 cmdstr[MAXSTRING];
-    char                 optstr[MAXSTRING];
-    short int            outfilenum;
-    char                 varname[MAXSTRING];
-    int                  outvarnum;
-    char                 format[MAXSTRING];
-    char                 typestr[MAXSTRING];
-    int                  type;
-    char                 multstr[MAXSTRING];
-    double               mult;
-    size_t               noutfiles;
+    extern option_struct       options;
+    extern global_param_struct global_param;
+
+    char                       cmdstr[MAXSTRING];
+    char                       optstr[MAXSTRING];
+    char                       flgstr[MAXSTRING];
+    short int                  streamnum;
+    char                       varname[MAXSTRING];
+    int                        outvarnum;
+    char                       format[MAXSTRING];
+    char                       typestr[MAXSTRING];
+    int                        type;
+    char                       multstr[MAXSTRING];
+    double                     mult;
+    size_t                     nstreams;
+    size_t                     nvars;
+    unsigned int               nextagg;
 
     strcpy(format, "*");
 
     /** Read through global control file to find output info **/
 
     fgets(cmdstr, MAXSTRING, gp);
 
-    outfilenum = -1;
+    streamnum = -1;
     outvarnum = 0;
 
     // Count the number of output files listed in the global param file
-    noutfiles = count_n_outfiles(gp);
+    nstreams = count_n_outfiles(gp);
 
     // only parse the output info if there are output files to parse
-    if (noutfiles > 0) {
-        options.Noutfiles = noutfiles;
+    if (nstreams > 0) {
+        options.Noutstreams = nstreams;
 
-        *out_data_files = calloc(options.Noutfiles, sizeof(*(*out_data_files)));
-        if (*out_data_files == NULL) {
+        output_streams = calloc(options.Noutstreams, sizeof(*(output_streams)));
+        if (output_streams == NULL) {
+            log_err("Memory allocation error in parse_output_info().");
+        }
+        out_data_files = calloc(options.Noutstreams, sizeof(*(out_data_files)));
```

Comment:
The previous implementation assigned the allocated space to `*out_data_files`.  The reason for that was that out_data_files is only 1-dimensional.  Its address was passed to this function so that the allocated space was stored in the original copy of the pointer in the parent function to this one.  So, to allocate out_data_files here, you have to dereference it to get back to a 1-d pointer.  I bet this is related to the memory issues you're encountering, although I haven't looked at the freeing of this array.  I'm surprised it would even function correctly though...  Let me keep looking through the code...